index_api-v2.md

```diff
@@ -0,0 +1,191 @@
+
+They need to store their profile at a publicly accessible URL (`profile_url`), and then submit the `profile_url` to the index.
+
+> :warning: SCHEMA REQUIREMENT
+>
+> A list of `linked_schemas` (array of strings) must be included in every profile (and therefore every schema) as a required field.
+>
+> ```json
+> {
+>   "linkedSchemas": ["test_schema-v2.1.0", "demo-v1"]
+> }
+> ```
+>
+> The library should never accept schemas and the index should never accept profiles without the `linked_schemas` property.
+
+#### Input
+
+- Profile that validates to a referenced schema (or list of schemas), available at a publicly accessible URL
+  - The profile must be available at the `profile_url` or it can't be recorded by the index
+  - The profile must include one or more `linked_schemas` (unique schema name(s) within a namespace) against which the profile must be validated
+
+#### Output
+
+- `node_id` - hash of the `profile_url`
+
+> :construction: HASH ALGORITHM
+>
+> The current implementation of the index uses the SHA256 hashing algorithm which produces a 64-character output. The purpose of hashing the `profile_url` is to make it easy to reference as a path parameter when requesting information about the node from the index (e.g., `GET /nodes/{node_id}` as described below).
+
+#### Sequence Diagram - Add Node Profile to Index
+
+![Add Node Profile to Index](index_api-v1-diagram_1.png)
+
+### [`POST /nodes-sync`](https://app.swaggerhub.com/apis-docs/MurmurationsNetwork/IndexAPI/2.0.0#/Node%20Endpoints/post_nodes_sync)
+
+This endpoint is identical to the `POST /nodes-sync` endpoint above but it responds with an immediate success or failure status, so it is useful for posting nodes in an asynchronous call to the index.
+
+#### Input
+
+- Same as for the `POST /nodes` endpoint above
+
+#### Output
+
+- Same as for the `GET /nodes/{node_id}` endpoint below
+
+### [`GET /nodes/{node_id}`](https://app.swaggerhub.com/apis-docs/MurmurationsNetwork/IndexAPI/2.0.0#/Node%20Endpoints/get_nodes__node_id_)
+
+The record of a node in the index's database can be in one of five possible states: `received`, `validated`, `validation_failed`, `posted`, `post_failed` or `deleted`. The node will only be discoverable in the index when it has the status of `posted` or `deleted`.
+
+This endpoint enables a Murmurations profile generator to get and present an update to the node operator as to the status of the node profile after it has been submitted to the index (i.e., when using `POST /nodes`).
+
+#### Input
+
+- the `node_id` of the node's profile that is currently in the index
+
+#### Output
+
+Detailed response examples can be seen in the [Open API specification](https://app.swaggerhub.com/apis-docs/MurmurationsNetwork/IndexAPI/2.0.0#/Node%20Endpoints/get_nodes__node_id_) under the _Responses_ section for `200 OK` response code.
+
+##### Success
```

Comment:
Looks to me like maybe these success and error outputs should be under the POST /nodes endpoint, rather than GET /nodes ?

Reply:
No, this is correct. The `POST /nodes` endpoint only returns a 200 OK to confirm that the `profile_url` has been received along with a `node_id`. It is then put into a queue and processed, and the status of the processing can be found using the `GET /nodes/{node_id}` endpoint.

More explanation on this point below in response to your other related questions...

Reply:
Right. Makes sense. I was a bit thrown off thinking of the GET /nodes endpoint as a more typical endpoint for retrieving an item, and the error or success info here is referring to the results of the POST, not the GET. But, since I suppose this endpoint will only really be used to check the status of a node (rather than to actually get it for something), this makes sense.